sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClientIntegrationTest.java

```diff
@@ -654,7 +660,313 @@ void setAndGetSessionState(MessagingEntityType entityType) {
+            .setIsAutoCompleteEnabled(false);
+
+        // Blocking here because it is not part of the scenario we want to test.
+        sendMessage(messages).block(TIMEOUT);
+
+        ServiceBusReceivedMessageContext receivedMessage = receiver.receive(options).next().block(TIMEOUT);
+        assertNotNull(receivedMessage.getMessage());
+
+        logger.info("Received message. Seq: {}.", receivedMessage.getMessage().getSequenceNumber());
+
+        // Assert & Act
+        try {
+            ServiceBusReceivedMessageContext finalReceivedMessage1 = receivedMessage;
+            StepVerifier.create(Mono.delay(Duration.ofSeconds(10))
+                .then(Mono.defer(() -> receiver.renewSessionLock(finalReceivedMessage1.getSessionId()))))
+                .assertNext(lockedUntil -> {
+                    assertNotNull(lockedUntil, "Could not renew session lock.");
+                })
+                .verifyComplete();
+        } finally {
+            logger.info("Completing message. Seq: {}.", receivedMessage.getMessage().getSequenceNumber());
+
+            receiver.complete(receivedMessage.getMessage())
+                .doOnSuccess(aVoid -> messagesPending.decrementAndGet())
+                .block(TIMEOUT);
+        }
+
+        receivedMessage = receiver2.receive(options).next().block(TIMEOUT);
+        assertNotNull(receivedMessage);
+
+        try {
+            ServiceBusReceivedMessageContext finalReceivedMessage = receivedMessage;
+            StepVerifier.create(Mono.delay(Duration.ofSeconds(10))
+                .then(Mono.defer(() -> receiver2.renewSessionLock(finalReceivedMessage.getSessionId()))))
+                .assertNext(lockedUntil -> {
+                    assertNotNull(lockedUntil, "Could not renew session lock.");
+                })
+                .verifyComplete();
+        } finally {
+            logger.info("Completing message. Seq: {}.", receivedMessage.getMessage().getSequenceNumber());
+
+            receiver2.complete(receivedMessage.getMessage())
+                .doOnSuccess(aVoid -> messagesPending.decrementAndGet())
+                .block(TIMEOUT);
+        }
+    }
+
+    /**
+     * Verifies that we can send messaged with some delay and receive a message without receiver dying.
+     */
+    @MethodSource("messagingEntityWithSessions")
+    @ParameterizedTest
+    void receiveSlowMultipleMessageAutoComplete(MessagingEntityType entityType, boolean isSessionEnabled) {
+        // Arrange
+        int count = 3;
+        setSenderAndReceiver(entityType, isSessionEnabled);
+        Duration delayPublishing = Duration.ofMillis(1000);
+
+        final String messageId = UUID.randomUUID().toString();
+
+        Flux.just(
+            getMessage(messageId, isSessionEnabled),
+            getMessage(messageId, isSessionEnabled),
+            getMessage(messageId, isSessionEnabled)
+        )
+            .delayElements(delayPublishing)
+            .flatMap(message -> sender.send(message).then())
```

Comment:
if you do this through `sendMessage(message)`, you won't need to increment the count below. The count is incremented because we published messages.. so it shouldn't be at the end.